src/main/scala/where/Condition.scala

```diff
@@ -76,3 +77,12 @@ case class Expression(string: String, references: Reference*) extends Condition
   def identifiables: Set[Identifiable] =
     (references flatMap (_.getReferenceable)).toSet
 }
+
+case class NoRelationshipCondition(node: Node) extends Condition {
```

Comment:
@a-wortman nah, unless we wanted to make `Path` have a `where` method that specifically takes `HasNoRelationships`, but then it still doesn't work with multiple conditions, and i think it just gets too hairy for a special case. we *could* do this though:

```scala
node where node.hasNoRelationships returns node
```
(just make it a convenience method on the `Node` trait)